core/src/main/java/com/oath/oak/InternalOakMap.java

```diff
@@ -317,250 +340,240 @@ V put(K key, V value, Function<ByteBuffer, V> transformer) {
             throw new NullPointerException();
         }
 
-        Chunk<K, V> c = findChunk(key); // find chunk matching key
-        Chunk.LookUp lookUp = c.lookUp(key);
-        // If there is a matching value reference for the given key, and it is not marked as deleted, then this put
-        // changes the slice pointed by this value reference.
-        if (lookUp != null && lookUp.valueSlice != null) {
-            Result<V> res = ValueUtils.exchange(c, lookUp, value, transformer, valueSerializer, memoryManager);
-            if (res.operationResult == SUCCESS) {
-                return res.value;
+        while (true) {
```

Comment:
Is it all that you replaced the recursive call with a loop? Why so?

Reply:
It is easier on the call stack (less memory and less nested calls), and it is easier to debug if necessary (can follow the loop manually instead of having a StackOverflowException).